### PR TITLE
Fix BuildPythonAPI cleaning and success message

### LINKED ISSUE #2339: CARLA 0.9.7 does not generate a python library on Windows 10
Hi,

Running `make PythonAPI` claims to complete successfully with following output.

```
-[BuildPythonAPI]: Carla lib for python has been successfully installed in "C:/path/to/Workspace/Cpp/carla/PythonAPI\carladist"! 
```

At first sight, the mixed usage of path separators looks suspicious. However, even if I ignore the forward slashes (i.e., /) by replacing them with backwards slash (i.e.,\), there is no `carladist` in that path.

As user, I would expect there is an actual output in the highlighted path given with the build output.

# Test environment

Windows 10 1909
Visual Studio 2017
Unreal Engine 4.22.3
Python 3.7.4.final.0
GNU Make 4.2.1
cmake version 3.12.18081601-MSVC_2
carla 0.9.7 (b9fbbf7fd03ee2a4f3a7baf4343a28381cfe81ef)

### PR DESCRIPTION
**Pull request by jeffreykxiao**
#### Description

BuildPythonAPI.bat was broken because it was expecting `%PYTHON_LIB_PATH%` to have a trailing slash, which it was missing. 

Fixes #2339  

Behavior prior to this change:

- `make clean` does not clean the intermediate files for building PythonAPI
- `make PythonAPI` displays the wrong path for the build result

Behavior with this change:

- `make clean` properly cleans up the intermediate files located in `PythonAPI/carla/build`, `PythonAPI/carla/dist`, and `PythonAPI/carla/source/carla.egg-info`
- `make PythonAPI` reports the correct location of the build result

#### Where has this been tested?

  * **Platform(s):** Windows 10 x64
  * **Python version(s):** ... 3.6
  * **Unreal Engine version(s):** ... 4.22.3

#### Possible Drawbacks

Now that `make clean` actually attempts to clean up the intermediate build artifacts, in case they are held by some other process, `make clean` may fail to clean up some of the directories. This is bog standard for Windows so as long as people are paying attention to the output of `make clean` all should be well.

Existing issue: code does not check for existing installs, so we can still end up with multiple egg files in `dist` if we do not `make clean`

---
This change is [Reviewable](https://reviewable.io/reviews/carla-simulator/carla/2348)

### CODE REVIEW

File: Util/BuildTools/BuildPythonAPI.bat [hunk 1]
```diff
@@ -153,4 +153,4 @@ rem ============================================================================
 
 :bad_exit
     endlocal
-    exit /b %errorlevel%
+    exit /b %errorlevel%
```
Comment (gokhanbarisaker) [inline Util/BuildTools/BuildPythonAPI.bat]: Something trivial. But, It seems that you deleted the empty line at the end of the file.

```suggestion
    exit /b %errorlevel%
    
```

**Comment by veliona** [comment]
Hi @jeffreykxiao !
Just facing this problem, thanks for your PR 👍 
This fixes the displayed path, however, doesn't fix the fact that I have an error: 
`error: command 'C:\\Program Files (x86)\\Microsoft Visual Studio\\2017\\Community\\VC\\Tools\\MSVC\\14.16.27023\\bin\\HostX86\\x64\\link.exe' failed with exit status 1120`

Are you able to give a hand on this?

File: Util/BuildTools/BuildPythonAPI.bat (hunk 1 above)
Comment (jeffreykxiao) [reply Util/BuildTools/BuildPythonAPI.bat]: Hm, it seems I did. IDK if the whole repo attempts to maintain posix compliance but nonetheless unnecessary changes are bad so I will gladly accept this

**Comment by jeffreykxiao** [comment]
@veliona 
> `error: command 'C:\\Program Files (x86)\\Microsoft Visual Studio\\2017\\Community\\VC\\Tools\\MSVC\\14.16.27023\\bin\\HostX86\\x64\\link.exe' failed with exit status 1120`
> 
> And I am still missing `dist` directory with `.egg` file..

That problem is unrelated to this PR (which only attempts to fix a message + make clean functionality), so I think it would be best to create a new issue. Additionally when doing so, adding some more context would be great! Also don't forget that the Discord server is available to help in real time.

**Comment by marcgpuig** [comment]
@jeffreykxiao Thanks for your PR.
Looks good to me, I've not tested it since I have no Windows access right now, but I hope @bernatx can test it in his machine.
Cheers!

**Comment by jeffreykxiao** [comment]
Hi @bernatx, just wanted to ping you real quick about this one  since it's a fairly easy one just waiting for your testing / approval :)

**Comment by filippogandolfi** [comment]
Hi all, I have what it looks like the same problem as @veliona .

> This fixes the displayed path, however, doesn't fix the fact that I have an error:
> `error: command 'C:\\Program Files (x86)\\Microsoft Visual Studio\\2017\\Community\\VC\\Tools\\MSVC\\14.16.27023\\bin\\HostX86\\x64\\link.exe' failed with exit status 1120`
> 
> And I am still missing `dist` directory with `.egg` file..
> Are you able to give a hand on this?

 However I have the new version with the code reviewed by @jeffreykxiao  and I don't get why I have this problem.
Someone has the same problem?

**Comment by EchoYJJ** [comment]
> Hi all, I have what it looks like the same problem as @veliona .
> 
> > This fixes the displayed path, however, doesn't fix the fact that I have an error:
> > `error: command 'C:\\Program Files (x86)\\Microsoft Visual Studio\\2017\\Community\\VC\\Tools\\MSVC\\14.16.27023\\bin\\HostX86\\x64\\link.exe' failed with exit status 1120`
> > And I am still missing `dist` directory with `.egg` file..
> > Are you able to give a hand on this?
> 
> However I have the new version with the code reviewed by @jeffreykxiao and I don't get why I have this problem.
> Someone has the same problem?

Hi, I also encounter the same issue, have you solved it now?

**Pull request merged**